docs/integrations/pytorch.rst

```diff
@@ -3,109 +3,370 @@
+To do that you just need to install psutil.
+
+.. code-block:: bash
+
+    pip install psutil
+
+.. image:: ../_static/images/integrations/pytorch-hardware.png
+   :target: ../_static/images/integrations/pytorch-hardware.png
+   :alt: PyTorch hardware consumption charts
+
+|example-hardware|
+
+Log hyperparameters
+^^^^^^^^^^^^^^^^^^^
+You can log training and model hyperparameters.
+To do that just pass the parameter dictionary to ``neptune.create_experiment`` method:
 
-Full script
-^^^^^^^^^^^
 .. code-block:: python3
 
-    # imports
-    import neptune
-    import torch
-    import torch.nn as nn
-    import torch.nn.functional as F
-    import torch.optim as optim
-    from torchvision import datasets, transforms
-
-    # set project
-    neptune.init(api_token='ANONYMOUS',
-                 project_qualified_name='shared/pytorch-integration')
-
-    PARAMS = {'DEVICE': torch.device("cuda" if torch.cuda.is_available() else "cpu"),
-              'ITERATIONS': 1000,
-              'batch_size': 64,
-              'lr': 0.005,
-              'momentum': 0.9}
-
-    class Net(nn.Module):
-        def __init__(self):
-            super(Net, self).__init__()
-            self.conv1 = nn.Conv2d(1, 20, 5, 1)
-            self.conv2 = nn.Conv2d(20, 50, 5, 1)
-            self.fc1 = nn.Linear(4*4*50, 500)
-            self.fc2 = nn.Linear(500, 10)
-
-        def forward(self, x):
-            x = F.relu(self.conv1(x))
-            x = F.max_pool2d(x, 2, 2)
-            x = F.relu(self.conv2(x))
-            x = F.max_pool2d(x, 2, 2)
-            x = x.view(-1, 4*4*50)
-            x = F.relu(self.fc1(x))
-            x = self.fc2(x)
-            return F.log_softmax(x, dim=1)
-
-    # start experiment
-    neptune.create_experiment(name='pytorch-example', params=PARAMS)
-
-    train_loader = torch.utils.data.DataLoader(
-        datasets.MNIST('../data',
-                       train=True,
-                       download=True,
-                       transform=transforms.Compose([transforms.ToTensor(),
-                                                     transforms.Normalize((0.1307,), (0.3081,))])),
-       batch_size=PARAMS['batch_size'],
-       shuffle=True)
-
-    model = Net().to(PARAMS['DEVICE'])
+    PARAMS = {'lr':0.005,
+               'momentum':0.9,
+               'iterations':100}
 
     optimizer = optim.SGD(model.parameters(), PARAMS['lr'], PARAMS['momentum'])
 
+    # log params
+    neptune.create_experiment('pytorch-advanced', params=PARAMS)
+
+.. image:: ../_static/images/integrations/pytorch-parameters.png
+   :target: ../_static/images/integrations/pytorch-parameters.png
+   :alt: PyTorch hyperparameter logging
+
+|example-hyperparameters|
+
+Log model weights
+^^^^^^^^^^^^^^^^^
+You can log model weights to Neptune both during and after training.
+
+To do that just use a ``neptune.log_artifact`` method on the saved model file.
```

Comment:
If you do `:meth:method` not ``method``, then users can click on it as see docstrings.